subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/KotlinBuildScript.kt

```diff
@@ -46,6 +46,8 @@ annotation class KotlinScriptTemplate
     resolver = KotlinBuildScriptDependenciesResolver::class,
     scriptFilePattern = ".*\\.gradle\\.kts")
 @ScriptTemplateAdditionalCompilerArguments([
+    "-java-parameters",
```

Comment:
❓ I don't understand why the new compiler options are also required in the script templates. I understood `-Xuse-old-class-files-reading` was required to get the compiler to load parameter names from the transformed Gradle api jar and `-java-parameters` was about preserving parameter names in the emitted class files, none of which seem to apply to the scripts, what am I missing?

Reply:
Right. I've been overzealous. We don't need those for script compilation. And we don't need `-Xuse-old-class-files-reading` on the `kotlin-dsl` plugin.